### PR TITLE
feat: generate action test mode

### PR DESCRIPTION
**Pull request by ryan-timothy-albert**
https://linear.app/speakeasy/issue/SPE-4059/feature-verify-validation-and-target-regeneration-in-github-actions

A request has come up a few times from customers to provide a test mode for our generation github action that does the following
- validates
- generates
- compiles
- tests correct outputs, releases

But does not write to github or have any other side-effects

### CODE REVIEW

File: internal/actions/outputs.go [hunk 1]
```diff
@@ -44,8 +46,11 @@ func setOutputs(outputs map[string]string) error {
 
 func printAndWriteString(f *os.File, out string) error {
 	fmt.Print(out)
-	if _, err := f.WriteString(out); err != nil {
-		return fmt.Errorf("error writing output: %w", err)
+	// We only want to print outputs if we are in test mode
```
Comment (chase-crumbaugh) [inline internal/actions/outputs.go:49]: This comment seems wrong
Also why would we not want to print the outputs in test mode?

File: internal/actions/runWorkflow.go [hunk 2]
```diff
@@ -72,7 +72,7 @@ func RunWorkflow() error {
 
 	success := false
 	defer func() {
-		if (!success || environment.GetMode() == environment.ModeDirect) && !environment.IsDebugMode() {
+		if (!success || environment.GetMode() == environment.ModeDirect) && !environment.IsDebugMode() && mode != environment.ModeTest {
```
Comment (chase-crumbaugh) [inline internal/actions/runWorkflow.go:75]: This condition is out of control, can we make it more readable?

File: internal/actions/runWorkflow.go [hunk 3]
```diff
@@ -265,6 +272,27 @@ func finalize(inputs finalizeInputs) error {
 			}
 		}
 
+		inputs.Outputs["commit_hash"] = commitHash
+	case environment.ModeTest:
```
Comment (chase-crumbaugh) [inline internal/actions/runWorkflow.go:276]: I thought we were exiting before calling finalize? Line 169

File: internal/actions/outputs.go (hunk 1 above)
Comment (ryan-timothy-albert) [reply internal/actions/outputs.go:49]: Oh I think I just worded it confusingly.

We always print outputs. We don't actually write the outputs in test mode

File: internal/actions/runWorkflow.go (hunk 2 above)
Comment (ryan-timothy-albert) [reply internal/actions/runWorkflow.go:75]: Ya. Will add a function

File: internal/actions/runWorkflow.go (hunk 3 above)
Comment (ryan-timothy-albert) [reply internal/actions/runWorkflow.go:276]: Ya true I changed that later. This is redundant

**Review by chase-crumbaugh — APPROVED** [review]

**Pull request merged**